mautic/mautic
# Bulk importing custom fields

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [x] running with no args throws a deprecation so we can make `--all` the default behavior in a future version.
| BC breaks? (use the c.x branch)        | [x]
| Automated tests included?              | [x] 
| Related user documentation PR URL      | https://github.com/mautic/user-documentation/pull/183
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

This adds an option to the `mautic:custom-field:create-column` command to not specify a field ID. Without the ID specified, it will create all remaining custom fields in one go.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Set `'create_custom_field_in_background' => true,` to your `app/config/local.php` file
3. In the database insert multiple lead fields with `column_is_not_created` set to 1
4. Run `bin/console mautic:custom-field:create-column`
5. Check that the fields have been created
